### PR TITLE
Update to LLVM 13.0.1

### PR DESCRIPTION
**Pull request by kmod**
I was running into an optimization problem with one of our traces, and tracked it down to something that was fixed in newer versions of LLVM. So I decided to update our LLVM from 10.0 to 13.0.1 (latest release). The main issues:
- ORC v1 was removed, so I had to migrate to ORC v2
- We had some bugs that were not triggered by the IR that LLVM 10.0 generates, but are triggered by LLVM 13

### CODE REVIEW

**Comment by kmod** [comment]
The PR didn't trigger CI so I tried marking it "ready for review", but the issue was that there was a merge conflict. I don't see how to set it back as a draft -- this isn't actually ready for review.

**Comment by kmod** [comment]
Ok this is now ready for review!

File: .github/workflows/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:latest
+FROM ubuntu:20.04
```
Comment (undingen) [inline .github/workflows/Dockerfile:1]: good catch!

File: pyston/pystol/pystol.cpp [hunk 2]
```diff
@@ -262,8 +262,7 @@ bool PystolFactDeriver::deriveFacts(Value* v, FactSet& facts, LLVMEvaluator& eva
             } else if (name == "PyTuple_New" || name == "PyTuple_New_Nonzeroed") {
                 RELEASE_ASSERT(isPyObjectPtr(call->getType()), "");
 
-                Type* t = call->getType();
-                t = call->getParent()->getParent()->getParent()->getTypeByName("struct._typeobject")->getPointerTo();
+                Type* t = StructType::getTypeByName(getContext(), "struct._typeobject");
```
Comment (undingen) [inline pyston/pystol/pystol.cpp:265]: :smile:

**Review by undingen — APPROVED** [review]
yeah very nice :sparkler:

**Pull request merged**